File: src/main/java/leetcode/offer/Offer10_1.java
package leetcode.offer;

/**
 * @author jingxinwu
 * @date 2021-11-19 10:08 下午
 */
public class Offer10_1 {

    /**
     * 输入：n = 5
     * 输出：5
     */


    public static void main(String[] args) {
        Offer10_1 offer10_1 = new Offer10_1();
        System.out.println(offer10_1.fib(9));
    }

    public int fib(int n) {

        if (n == 0 || n == 1) {
            return n;
        }
        int[] dp = new int[n];
        dp[0] = 1;
        dp[1] = 1;
        for (int i = 2; i < n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[dp.length - 1] % 1000000007;
    }
}

File: src/main/go/leetcode/lists/offer/Offer10_1.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/28 21:22
 */

func fib(n int) int {
	if n == 0 {
		return 0
	}
	var first int64
	var second int64
	first, second = 0, 1
	for i := 1; i <= n; i++ {
		t := first
		first = second
		second = (t + second) % 1000000007
	}
	return int(first % 1000000007)
}
